packages/internal-helpers/src/remote.ts

```diff
@@ -62,6 +99,16 @@ export function matchPathname(url: URL, pathname?: string, allowWildcard?: boole
 	return false;
 }
 
+/**
+ * Determines whether a given remote resource, identified by its source URL,
+ * is allowed based on specified domains and remote patterns.
+ *
+ * @param {string} src - The source URL of the remote resource to be validated.
+ * @param {Object} options - The configuration options for domain and pattern matching.
+ * @param {string[]} options.domains - A list of allowed domain names.
+ * @param {RemotePattern[]} options.remotePatterns - A list of allowed remote patterns for matching.
+ * @return {boolean} Returns true if the source URL matches any of the specified domains or remote patterns; otherwise, false.
```

Comment:
```suggestion
 * @return {boolean} Returns `true` if the source URL matches any of the specified domains or remote patterns; otherwise, `false`.
```